python/vote-text-bot/bot.py

```diff
@@ -0,0 +1,120 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+'''
+Hangouts Chat simple text-only vote bot in Python App Engine
+
+@see developers.google.com/hangouts/chat/how-tos/cards-onclick
+'''
+
+import json
+import webapp2
+
+def createMessage(voter, vote_count=0, update=False):
+    """
+    Create and return new interactive card or update existing one
+
+    @type  voter:       str
+    @param voter:       user (display) name
+    @type  vote_count:  int
+    @param vote_count:  current vote count
+    @type  update:      bool
+    @param update:      update existing or create new card?
+    @rtype:             dict
+    @return:            JSON payload to return to Hangouts Chat
+    @see:   developers.google.com/hangouts/chat/concepts/cards
+    """
+
+    PARAMETERS = [{'key': 'count', 'value': str(vote_count)}]
+    return {
+        'actionResponse': {
+            'type': 'UPDATE_MESSAGE' if update else 'NEW_MESSAGE'
+        },
+        'cards': [{
+            'header': {'title': 'Last vote by %s!' % voter},
+            'sections': [{
+                'widgets': [{
+                    'textParagraph': {'text': '%d votes!' % vote_count}
+                }, {
+                    'buttons': [{
+                        'textButton': {
+                            'text': '+1',
+                            'onClick': {
+                                'action': {
+                                    'actionMethodName': 'upvote',
+                                    'parameters': PARAMETERS,
+                                }
+                            }
+                        }
+                    }, {
+                        'textButton': {
+                            'text': '-1',
+                            'onClick': {
+                                'action': {
+                                    'actionMethodName': 'downvote',
+                                    'parameters': PARAMETERS,
+                                }
+                            }
+                        }
+                    }, {
+                        'textButton': {
+                            'text': 'NEW',
+                            'onClick': {
+                                'action': {
+                                    'actionMethodName': 'newvote',
+                                }
+                            }
+                        }
+                    }]
+                }]
+            }]
+        }]
+    }
+
+class MainPage(webapp2.RequestHandler):
```

Comment:
`MainPage` doesn't really make sense for this server. Rename the server class.

Reply:
DONE